packages/expo-video/ios/NowPlayingManager.swift

```diff
@@ -171,7 +171,18 @@ class NowPlayingManager: VideoPlayerObserverDelegate {
       nowPlayingInfo[MPNowPlayingInfoPropertyIsLiveStream] = currentItem.duration.isIndefinite
       nowPlayingInfo[MPNowPlayingInfoPropertyPlaybackRate] = await player.rate
       nowPlayingInfo[MPNowPlayingInfoPropertyMediaType] = MPNowPlayingInfoMediaType.video.rawValue // Using MPNowPlayingInfoMediaType.video causes a crash
-      nowPlayingInfo[MPMediaItemPropertyArtwork] = artwork
+      if let artworkURLString = userMetadata?.artwork,
+      let artworkURL = URL(string: artworkURLString),
+      let imageData = try? Data(contentsOf: artworkURL),
+      let image = UIImage(data: imageData) {
+        let artwork = MPMediaItemArtwork(boundsSize: image.size) { _ in
+          return image
+        }
+        nowPlayingInfo[MPMediaItemPropertyArtwork] = artwork
+      } else if let artwork = artwork {
+        // If no URL string or error occurs, use the fallback artwork
+        nowPlayingInfo[MPMediaItemPropertyArtwork] = artwork
+      }
```

Comment:
This will cause issues. For videos with large artworks it will take a while to fetch on slow networks, therefore if someone skips to next video quickly we will have multiple Tasks racing to replace the artwork and the older, slower one will overwrite the new, faster one while we want to display the new one.

Additionally we might end up with a few data data requests while we only ever need one at a time. 

This function also might be called multiple times, so we have to make sure that we don't needlessly re-fetch the artwork.

We can implement a data task based solution.
First lets add a variable on the top of the file under the players variable:
```swift
private var players = NSHashTable<VideoPlayer>.weakObjects()
private var artworkDataTask: URLSessionDataTask?
```

now let's replace this part with 
```suggestion
      if let artworkUrlString = userMetadata?.artwork, let url = URL(string: artworkUrlString), artworkDataTask?.originalRequest?.url != url {
        artworkDataTask?.cancel()
        artworkDataTask = fetchArtwork(url: url) { image in
          // We can't reuse the `nowPlayingInfo` as the actual nowPlayingInfo might've changed while the image was being fetched
          var currentNowPlayingInfo = MPNowPlayingInfoCenter.default().nowPlayingInfo ?? [:]
          currentNowPlayingInfo[MPMediaItemPropertyArtwork] = image
          MPNowPlayingInfoCenter.default().nowPlayingInfo = currentNowPlayingInfo
        }
      } else if userMetadata?.artwork == nil {
        self.artworkDataTask = nil
        nowPlayingInfo[MPMediaItemPropertyArtwork] = artwork
      }

```

and add the function at the bottom of the class:

```swift
  private func fetchArtwork(url: URL, completion: @escaping (MPMediaItemArtwork?) -> Void) -> URLSessionDataTask {
    let task = URLSession.shared.dataTask(with: url) { data, response, error in
      if let error = error {
        log.warn("ExpoVideo - Couldn't fetch the artwork: \(error.localizedDescription)")
        completion(nil)
        return
      }

      guard let data, response as? HTTPURLResponse != nil else {
        log.warn("ExpoVideo - Couldn't display the artwork: the response was empty")
        completion(nil)
        return
      }

      if let image = UIImage(data: data) {
        let artwork = MPMediaItemArtwork(boundsSize: image.size) { _ in
          return image
        }
        completion(artwork)
      } else {
        completion(nil)
      }
    }

    task.resume()
    return task
  }
```